Model/Charge.php

```diff
@@ -165,6 +169,30 @@ public function charge()
                 $this->_orderPaymentRepository->save($payment);
             }
 
+            // checking charge amount is same as order grand total
+            if ($charge->getAmount() != $this->_order->getGrandTotal()) {
+                $messageWithAmount = $this->_helper->__("Charge Amount is not same as Order amount. Charge amount:- %s and order amount :-%s", $charge->getAmount(),$this->_order->getGrandTotal());
+                $this->_logger->debug($messageWithAmount);
+                $this->_helper->cancelOrder($this->_order, $messageWithAmount);
+                $apiConfig = new \Zip\ZipPayment\MerchantApi\Lib\Configuration();
+                $apiConfig->setApiKey('Authorization', $this->_config->getMerchantPrivateKey())
+                    ->setApiKeyPrefix('Authorization', 'Bearer')
+                    ->setEnvironment($this->_config->getEnvironment())
+                    ->setPlatform("Magento/" . $this->_helper->getMagentoVersion()
+                        . "Zip_ZipPayment/"
+                        . $this->_helper->getExtensionVersion());
+                $payload = $this->_payloadHelper->getRefundPayload(
+                    $this->_order,
+                    $charge->getAmount(),
+                    $messageWithAmount
+                );
+                // doing refund if the amount is captured by the charge api
```

Comment:
Is there a check that the charge has been made? as this looks like will call refund irrespective?  Is this the correct/safe behavior?